### ISSUE TITLE
from_df error

### ISSUE THREAD

**Issue by jeweinberg** [Esri/arcgis-python-api#886]
The from_df function throws "key error" when the index of a dataframe doesn't start at 0 and/or skips integers. When slicing a dataframe in pandas it is common to have an index that does not start at 0 and may be missing some index keys or for the index to not be an integer at all. You are using .loc which is not the integer indexer for pandas. You should instead use iloc.

pd.DataFrame.spatial.from_df(df, address_column='address_txt', geocoder=geocode_obj)

**Comment by priyankatuteja** [comment]
@jeweinberg Thanks for reporting it. 

@achapkowski Could you check if this needs to be fixed.

code to reproduce:
```
from arcgis.gis import GIS
import pandas as pd
gis = GIS("https://geosaurus.maps.arcgis.com", "arcgis_python", "P@ssword123")

item=gis.content.get('b864539b1f3342cbab716c94742a8d6d')
item

sdf = item.layers[0].query(as_df=True)

sdf.drop(columns='SHAPE', inplace=True)

sdf.set_index('Point_Count', inplace=True)

df = sdf[:10]
pd.DataFrame.spatial.from_df(df, address_column='STATE')
```

error:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\indexes\base.py in get_loc(self, key, method, tolerance)
   2888             try:
-> 2889                 return self._engine.get_loc(casted_key)
   2890             except KeyError as err:

pandas\_libs\index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas\_libs\index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas\_libs\index.pyx in pandas._libs.index.IndexEngine._get_loc_duplicates()

pandas\_libs\index_class_helper.pxi in pandas._libs.index.Int64Engine._maybe_get_bool_indexer()

pandas\_libs\index.pyx in pandas._libs.index.IndexEngine._unpack_bool_indexer()

KeyError: 0

The above exception was the direct cause of the following exception:

KeyError                                  Traceback (most recent call last)
C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\arcgis\features\geo\_accessor.py in from_df(df, address_column, geocoder, sr, geometry_column)
   2258                         try:
-> 2259                             address = df.loc[start + index, address_column]
   2260                         except: # for older versions, fall back to `df.ix`

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\indexing.py in __getitem__(self, key)
    872                     pass
--> 873             return self._getitem_tuple(key)
    874         else:

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\indexing.py in _getitem_tuple(self, tup)
   1043         try:
-> 1044             return self._getitem_lowerdim(tup)
   1045         except IndexingError:

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\indexing.py in _getitem_lowerdim(self, tup)
    785                 #  caught by the _is_nested_tuple_indexer check above.
--> 786                 section = self._getitem_axis(key, axis=i)
    787 

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\indexing.py in _getitem_axis(self, key, axis)
   1109         self._validate_key(key, axis)
-> 1110         return self._get_label(key, axis=axis)
   1111 

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\indexing.py in _get_label(self, label, axis)
   1058         # GH#5667 this will fail if the label is not present in the axis.
-> 1059         return self.obj.xs(label, axis=axis)
   1060 

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\generic.py in xs(self, key, axis, level, drop_level)
   3481         else:
-> 3482             loc = self.index.get_loc(key)
   3483 

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\indexes\base.py in get_loc(self, key, method, tolerance)
   2890             except KeyError as err:
-> 2891                 raise KeyError(key) from err
   2892 

KeyError: 0

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
<ipython-input-29-190b631db751> in <module>
     15 
     16 df = sdf[:10]
---> 17 pd.DataFrame.spatial.from_df(df, address_column='STATE')

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\arcgis\features\geo\_accessor.py in from_df(df, address_column, geocoder, sr, geometry_column)
   2259                             address = df.loc[start + index, address_column]
   2260                         except: # for older versions, fall back to `df.ix`
-> 2261                             address = df.ix[start + index, address_column]
   2262                         try:
   2263                             loc = res[index]['location']

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\pandas\core\generic.py in __getattr__(self, name)
   5128             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5129                 return self[name]
-> 5130             return object.__getattribute__(self, name)
   5131 
   5132     def __setattr__(self, name: str, value) -> None:

AttributeError: 'DataFrame' object has no attribute 'ix'

```

**Comment by Detteor** [comment]
@achapkowski @jeweinberg 

I was having the same error when attempting to geocode a bunch of addresses, so I edited my _accessor.py file as such:
``` Python 
else:
           if geocoder is None:
                geocoder = arcgis.env.active_gis._tools.geocoders[0]
            if sr == None:
                sr = dict(geocoder.properties.spatialReference) """Forced it to use the inputted
sr otherwise it would use the default sr available to the geocoder"""
            geoms = []
            if address_column in df.columns:
                batch_size = geocoder.properties.locatorProperties.MaxBatchSize
                N = len(df)
                geoms = []
                for i in range(0, N, batch_size):
                    start = i
                    stop = i + batch_size if i + batch_size < N else N
                    res = batch_geocode(list(df[start:stop][address_column]), geocoder=geocoder)
                for index in range(len(res)):
                    try:
                        address = df.loc[start:index, address_column] #Changed from single value to index.
                    except: # for older versions, fall back to `df.ix`
                        address = df.ix[start:index, address_column]
                    try:
                        loc = res[index]['location']
                        x = loc['x']
                        y = loc['y']
                        geoms.append(Geometry({'x': x, 'y': y, 'spatialReference': sr}))

                    except:
                        x, y = None, None
                        try:
                            loc = geocode(address, geocoder=geocoder)[0]['location']
                            x = loc['x']
                            y = loc['y']
                        except:
                            print('Unable to geocode address: ' + address)
                            pass
                            geoms.append(None)
            else:
                raise ValueError("Address column not found in dataframe")
            df['SHAPE'] = geoms
            df.spatial.set_geometry("SHAPE")
            return df 
```

Sorry for all the extra whitespace.

**Comment by mohi9282** [comment]
A fix for this issue was implemented with Python API v1.9.0.
